### DISCUSSION TITLE
How to update only settings and configuration without full system update

### DISCUSSION THREAD

**Discussion by kunaljee12** [rauc/rauc#1581] [Q&A]
1. How to update only settings and configuration without full system update .

2. Only application update is possible without reboot

3. How to do Containerized application update

**Comment by jluebbe** [comment]
> 1. How to update only settings and configuration without full system update

RAUC is an image-based updater. If you configuration is stored in static files in the image, RAUC can only replace the full image. With adaptive updates and streaming, you can reduce the amount of data to download. Image-based updating is a critical part of how RAUC provides a robust updates.

For configuration changes at *runtime*, you application could generate config files in `/run` or in a data partition, but that's very project specific and out of scope for RAUC.

> 2. Only application update is possible without reboot
> 3. How to do Containerized application update

Take a look at https://github.com/rauc/rauc/issues/969 for some background with regards to updating decoupled applications or containers. The current development version of RAUC contains support for artifact updates:

- https://rauc.readthedocs.io/en/latest/changes.html#release-1-13-development
- https://rauc.readthedocs.io/en/latest/basic.html#sec-basic-artifact-repositories
- https://rauc.readthedocs.io/en/latest/integration.html#sec-repository-config

We'd be interested in feedback regarding this new approach.

**Reply by kunaljee12** [reply]
OK So what i understood from the above discussion , 

We need to keep all setting configuration and all application , containerised application and all the rootfilesystem and boot loader in a single artifact or bundle and after update we need to reboot every time Right ?

**Reply by kunaljee12** [reply]
Thanks for your quick reply.

Could you please help me out on below query :

1. Do we need to include all the settings, configurations, applications, containerized applications, root filesystem, and boot loader in a single artifact or bundle?
2. After updating everything, do we have to reboot the system every time?
3. Is it necessary to package all settings, configurations, and applications (including containerized ones) into one bundle for updates?
4. Does the update process require a reboot each time we update the system or applications?
5. Are we required to keep everything, such as the root filesystem and boot loader, in a single bundle for updates?
6 .Does updating the full system always require a reboot with RAUC?
7 .Can we update just one part of the system, like applications, without needing a full reboot?
8. Is it true that RAUC updates need to include all components and always require a reboot afterward?

Regards ,

kunal kumar

**Reply by ejoerns** [reply]
> Thanks for your quick reply.
> 
> Could you please help me out on below query :
> 
> 1. Do we need to include all the settings, configurations, applications, containerized applications, root filesystem, and boot loader in a single artifact or bundle?

No. Artifacts have the concept of repositories where each repository can have multiple artifacts and you can also have multiple repositories.  Depending on your needs. The base system is typically abstracted in conventional rauc 'slots'. A bundle may have only images or only artifacts or both artifacts and images.
Please refer to the referenced documentation for more details.

> 2. After updating everything, do we have to reboot the system every time?

For slots (or the base system) you typically need to reboot. For artifacts typically not. Depends on how you start you applications/containers. You can use a post-install hook to restart them for example.

> 3. Is it necessary to package all settings, configurations, and applications (including containerized ones) into one bundle for updates?

Looks like a repetition of 1.

> 4. Does the update process require a reboot each time we update the system or applications?

See my answer on 2.

> 5. Are we required to keep everything, such as the root filesystem and boot loader, in a single bundle for updates?
>    6 .Does updating the full system always require a reboot with RAUC?
>    7 .Can we update just one part of the system, like applications, without needing a full reboot?
> 6. Is it true that RAUC updates need to include all components and always require a reboot afterward?

Since questions repeat from here or are slight variations of the original questions, I'll close this discussion.

> Regards ,
> 
> kunal kumar

**Comment by leon-anavi** [comment]
Hi @kunaljee12,

I see that @jluebbe has already provided an in-depth answer. I am doing copy&paste of something that I was planning to post to your original request in meta-rauc-community:

Please note that meta-rauc-community is a git repository for demo integrations of RAUC on popular hardware platforms. Because of this there are other repositories within the RAUC Git organization which are more suitable for discussing RAUC features.

> 1. How to update only settings and configuration without full system update .

RAUC has an advanced feature called "Adaptive Updates". This feature allows RAUC to download only the parts of the bundle that are actually needed. Learn more about it in the documentation: https://rauc.readthedocs.io/en/latest/advanced.html#adaptive-updates

> 2. Only application update is possible without reboot

You can considering using containers for applications and keeping in mind your third question you have been already thinking about it. You can move some of the containers to a persistent /data partition that is used for storage in a typical RAUC integration.

> 3. How to do Containerized application update

You can use Yocto/OpenEmbedded layer [meta-virtualization](https://git.yoctoproject.org/meta-virtualization)
